src/unix/haiku/mod.rs

```diff
@@ -352,6 +352,66 @@ s_no_extra_traits! {
         __unused1: *mut ::c_void, // actually a function pointer
         pub sigev_notify_attributes: *mut ::pthread_attr_t,
     }
+
+    #[cfg(target_arch = "x86_64")]
+    pub struct fpu_state {
+        pub control: ::c_ushort,
+        pub status: ::c_ushort,
+        pub tag: ::c_ushort,
+        pub opcode: ::c_ushort,
+        pub rip: ::c_ulong,
+        pub rdp: ::c_ulong,
+        pub mxcsr: ::c_uint,
+        pub mxcsr_mask: ::c_uint,
+        _fpreg: [::c_uchar; 16],
```

Comment:
Is this correct?
The definition in headers/posix/arch/x86_64/signal.h is:
```
	union {
		struct x86_64_fp_register fp[8];
		struct x86_64_fp_register mmx[8];
	};
```

And then the definition of a `x86_64_fp_register` is:

```
struct x86_64_fp_register {
	unsigned char value[10];
	unsigned char reserved[6];
};
```
So it seems that the size should be 8*16 bytes?